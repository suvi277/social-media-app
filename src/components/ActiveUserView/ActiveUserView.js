import React from 'react';
import { Link } from 'react-router-dom';
import image from './profile.png';

const ActiveUserView = ({ email, firstName, lastName, id }) => {
	return (
		<div className="user-details d-flex">
			<img width={100} height={100} src={image} alt="profile" title="profile image" />
			<div className="px-3 py-2">
				<Link to={`/profile/${id}`}>
					<div className="title">{`${firstName} ${lastName}`}</div>
				</Link>
				<span className="email-id font-weight-lighter">{email}</span>
			</div>
		</div>
	);
};

export { ActiveUserView };
