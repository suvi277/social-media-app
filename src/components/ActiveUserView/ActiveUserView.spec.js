import React from 'react';
import { shallow } from 'enzyme';
import { ActiveUserView } from './ActiveUserView';

describe('ActiveUserView Component', () => {
	const mockUser = {
		id: 1,
		firstName: 'First',
		lastName: 'Last',
		email: 'first@gmail.com'
	};
	const wrapper = shallow(<ActiveUserView {...mockUser} />);

	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('should render a title className div', () => {
		expect(wrapper.find('.title')).toHaveLength(1);
	});
	it('should show the firstName and lastName in the title', () => {
		expect(wrapper.find('.title').text()).toBe(`${mockUser.firstName} ${mockUser.lastName}`);
	});
	it('should show the email in the email-id', () => {
		expect(wrapper.find('.email-id').text()).toBe(mockUser.email);
	});
});
