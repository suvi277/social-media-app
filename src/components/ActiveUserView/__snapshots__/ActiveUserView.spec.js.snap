// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActiveUserView Component renders as expected 1`] = `ShallowWrapper {}`;
